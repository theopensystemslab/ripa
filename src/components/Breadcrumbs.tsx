import Box from "@material-ui/core/Box";
import Container from "@material-ui/core/Container";
import * as React from "react";

import Stepper from "./Stepper";

interface IBreadcrumbs {
  active?: number;
  list?: string[];
  numbered?: boolean;
  Divider?: any;
}

const Breadcrumbs: React.FC<IBreadcrumbs> = ({
  active = 1,
  list = [],
  numbered = false,
  Divider = false
}) => {
  return (
    <Box bgcolor="background.paper" fontSize="h6.fontSize" py={4}>
      <Container maxWidth="lg">
        <Stepper
          active={active}
          list={list}
          numbered={numbered}
          Divider={Divider}
        />
      </Container>
    </Box>
  );
};

export default Breadcrumbs;
