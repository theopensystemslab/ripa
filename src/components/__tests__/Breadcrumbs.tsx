import { cleanup, render } from "@testing-library/react";
import * as React from "react";

import Breadcrumbs from "../Breadcrumbs";

afterEach(cleanup);

it("renders items correctly", async () => {
  const list = [
    "Property location",
    "About the works",
    "Application",
    "Declaration",
    "Pay fee"
  ];
  const active = 2;
  const { getByText } = await render(
    <Breadcrumbs active={active} list={list} />
  );
  expect(getByText("Property location")).toBeInTheDocument();
});

it("renders snapshot", () => {
  const { asFragment } = render(<Breadcrumbs />);
  expect(asFragment()).toMatchSnapshot();
});
