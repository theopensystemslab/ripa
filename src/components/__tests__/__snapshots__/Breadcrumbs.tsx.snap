// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-76"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg"
    >
      <div
        class="MuiBox-root MuiBox-root-96 makeStyles-root-85"
      >
        <button
          aria-controls="mobile-menu"
          aria-haspopup="true"
          class="MuiButtonBase-root makeStyles-moreBtn-91"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
